### ISSUE TITLE
Question about unknow voxels in occupancy loss calculations 

### ISSUE THREAD

**Issue by kowloontong** [SPengLiang/OccupancyM3D#6]
First of all, thank you for your great work. Second, i have a question about unknow voxels in occupancy loss calculations:
In the paper, you mentioned that unknow voxels do not participate in the calculation(Section 3.2.2 last line).

>  Note that we do not consider
unknown voxels in both the occupancy labels and occupancy losses.

But i don't see any processing in the code:
```
#@File image_vfe.py
#@Func: occupancy_loss
gt_one_hot[bin_ind == indices_repeat] = 1.
gt_one_hot[bin_ind > indices_repeat] = 2.  # 2 represent unknow voxels ?

depth_target = bin_ray_depths(depth_maps, **disc_cfg, target=True)
cls_loss = self.focal_loss(frustum_features_prob, depth_target[:, :-1, ...])

#@File image_vfe.py
#@Func: focal_loss
def focal_loss(self, input, target, alpha=0.25, gamma=2.):
   pos_inds = target.eq(1).float()
   neg_inds = target.lt(1).float()   # need to ignored target.eq(2), AKA  unknow voxels?
   loss = 0
   pos_loss = torch.log(input) * torch.pow(1 - input, gamma) * pos_inds * alpha
   neg_loss = torch.log(1 - input) * torch.pow(input, gamma) * neg_inds * (1 - alpha)
   num_pos = pos_inds.float().sum()
   pos_loss = pos_loss.sum()
   neg_loss = neg_loss.sum()
   if num_pos == 0:
     loss = loss - neg_loss
   else:
     loss = loss - (pos_loss + neg_loss) / num_pos
   return loss.mean()

```

Am i missing something?

**Comment by SPengLiang** [comment]
Thanks for your interest. The focal loss only handles the voxels whose values are in the range[0, 1]. "target.lt(1)" refers to find to the voxel values lower than 1. Thus, voxel values filled with 2 (unknown voxels) do not participate in the loss calculation.

**Comment by kowloontong** [comment]
Answered my confusion, thanks a lot.
